### PR TITLE
Some fixes to mlme-disassociate

### PR DESCRIPTION
**Pull request by jryun**
Fix frame formatting and building for disassociation notification

Fix not saving coordinator information when associating.

Fix Deregistering disassociation request listener

### CODE REVIEW

File: net/mac802154/cfg.c [hunk 1]
```diff
@@ -445,11 +445,6 @@ ieee802154_deregister_disassoc_req_listener( struct wpan_phy *wpan_phy,
 {
 	int r;
 	struct ieee802154_local *local = wpan_phy_priv(wpan_phy);
-	BUG_ON( NULL == local );
```
Comment (jryun) [inline net/mac802154/cfg.c]: This shoudl not have been removed.

File: net/ieee802154/nl802154.c [hunk 2]
```diff
@@ -1334,12 +1334,17 @@ static void nl802154_assoc_cnf( struct genl_info *info, u16 assoc_short_address,
 	netdev = info->user_ptr[1];
 	wpan_dev = netdev->ieee802154_ptr;
 
+	netdev->netdev_ops->ndo_open(netdev);
```
Comment (cfriedt) [inline net/ieee802154/nl802154.c]: checks for null pointers? return values?

**Comment by cfriedt** [comment]
has conflicts

**Comment by jryun** [comment]
I can't seem to see these conflicts on github, it only mentions that there are conflicts that must be resolved, but doesn't show them

**Pull request merged**